src/Transport/RequestBuilder.php

```diff
@@ -105,4 +108,27 @@ public function build($method, $uri, $params = null)
 
 		return $request;
 	}
-}
\ No newline at end of file
+
+    /**
+     * @param string $baseUrl
+     */
+    public static function getLastRedirectUrl($baseUrl)
+    {
+        $ch = curl_init($baseUrl);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_HEADER, 1);
+        curl_exec($ch);
+
+        $lastUrl = curl_getinfo($ch, CURLINFO_EFFECTIVE_URL);
```

Comment:
Is this the only way of getting the latest url for signing by making a separate curl call? Will increase latency a bit, right?

Reply:
Since we sign the request before sending the request, we need to make another request first to get the final url dynamically.